### PR TITLE
simplify dockerfile organization for lite/full images and upgrade dep…

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
…endencies on harvest and registry-mgr

## 🗒️ Summary

Simplify the Dockerfile with 2 targets for the Docker images Lite and Full.
Upgrade dependencies on harvest and registry-mgr for last integration tests before stable releases.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: __99_ %

## ⚙️ Test Data and/or Report

Using branch CICD for validation.

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -28,42 +28,36 @@
...
 # POSSIBILITY OF SUCH DAMAGE.
 
-FROM eclipse-temurin:25-jdk-jammy
+FROM eclipse-temurin:25-jdk-jammy AS base
 
-# Set following arguments with compatible versions
-ARG harvest_package_path=https://github.com/NASA-PDS/harvest/releases/download/v5.0.0/harvest-5.0.0-bin.tar.gz
-ARG reg_manager_package_path=https://github.com/NASA-PDS/registry-mgr/releases/download/v6.0.0/registry-manager-6.0.0-bin.tar.gz
-ARG test_data_url=https://pds.nasa.gov/data/pds4/test-data/registry/custom-datasets.tar.gz
+ARG harvest_package_path=https://github.com/NASA-PDS/harvest/releases/download/v5.1.0/harvest-5.1.0-bin.tar.gz
```
Comment (jordanpadams) [inline docker/Dockerfile]: @tloubrieu-jpl is there any way we can avoid hard coding these versions?

File: .github/workflows/branch-cicd.yaml [hunk 2]
```diff
@@ -12,10 +12,11 @@ name: 🔁 Branch integration testing
 # What event starts this workflow: a push to any branch other than main
 
 on:
-    push:
-        branches:
-            -   '**'
-            -   '!main'
+    # deactivate in favor of legacy-branch-cicd while testing with legacy harvest and registry-mgr repository
```
Comment (jordanpadams) [inline .github/workflows/branch-cicd.yaml]: @tloubrieu-jpl it may be better to just append .off or something to the filename so it does appear as an action at all? I think that’s what @nutjob4life has been doing or something similar?

**Review by jordanpadams — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (jordanpadams) [reply docker/Dockerfile]: If this is intended to be temporary we can merge and create a ticket to track removing this

Comment (tloubrieu-jpl) [reply docker/Dockerfile]: These values are default values which can be overriden when the docker build command is launched, see https://github.com/NASA-PDS/registry-loader/blob/46a9e39956a90d78603670be985212beb070fcd3/docker/README.md?plain=1#L28

That is what we will use with the single repository registry-loader, but with harvest and registry-mgr still in external repositories, we need a place to "save" what the dependencies are and that is what we used here.

I am hoping this build will be the latest harvest and registry-mgr releases from their own repositories.

File: .github/workflows/branch-cicd.yaml (hunk 2 above)
Comment (tloubrieu-jpl) [reply .github/workflows/branch-cicd.yaml]: I made that change. That sounds good. 

The docker/README.md that I forgot to update before.

Comment (nutjob4life) [reply .github/workflows/branch-cicd.yaml]: I think that's what I often do; it's certainly more "sure fire" than editing the triggers and wondering "did I turn off _all_ possible build triggers?" and then wondering "why am I getting an error saying that at least one trigger is required"?

**Review by jordanpadams — APPROVED** [review]
